tests/app/main/views/test_templates.py

```
@@ -776,6 +776,64 @@ def test_view_letter_template_does_not_display_send_button_if_template_over_10_p
     assert page.select_one("h1", {"data-error-type": "letter-too-long"})
 
 
+@pytest.mark.parametrize("template_page_count", [6, 10])
+@pytest.mark.parametrize("template_type", ["letter", "email", "sms"])
+def test_view_letter_template_has_attach_pages_button_if_template_below_10_pages_long(
```

I think there are some sticky edge cases that you should think about here as epic lead (but potentially could be handled separately). If the template is 9 pages long with the blank personalisation it could hit 10 pages with personalisation injected (esp if some of the personalisation is eg an address that spans multiple lines). How do we want to handle that?

I think the button disappearing altogether is maybe weird - possibly we show show the button but it flags an error saying 'you can't attach pages as the letter would be too long', or the button is greyed out, or something. But disappearing altogether might be odd?

---

> If the template is 9 pages long with the blank personalisation it could hit 10 pages with personalisation injected (esp if some of the personalisation is eg an address that spans multiple lines). How do we want to handle that?

That feels a bit out of scope. Also we don't do anything like this when letter template is exactly 10 pages long - we don't warn users it can easily go over based on placeholders.
I think as long as there is a space to add an attachment, we should offer that option, and then if the whole would breach 10 pages upon sending the letter due to placeholder values, we will throw an error then.

> I think the button disappearing altogether is maybe weird - possibly we show show the button but it flags an error saying 'you can't attach pages as the letter would be too long', or the button is greyed out, or something. But disappearing altogether might be odd?

Hmm... We could show a message instead a button, like: "this letter template is at maximum length, no space left for an attachment". If we do that it will make sense to split those test cases you mentioned, indeed. I will ask @karlchillmaid about the right content.

---

Also, one edge case you definitely need to have in mind is what happens if someone edits the template after attaching some pages and that pushes the page over the 10 pages limit.

---

Hmmm... I think the same should happen that we do when they try to edit the template to go over 10 pages by itself. Worth checking if we don't allow them to save such changes, or if we block them when they are sending.

---

I will add a ticket for this.

---

Voilla: https://trello.com/c/PqMqV88F/222-for-a-template-with-an-attachment-show-an-error-if-their-total-number-of-pages-is-too-long 😊